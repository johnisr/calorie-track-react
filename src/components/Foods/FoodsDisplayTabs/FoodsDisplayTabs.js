import React from 'react';
import { connect } from 'react-redux';
import {Tabs, TabList, Tab, TabPanels, TabPanel} from '../../Tabs/Tabs';
import EditFoodFormForFoods from './EditFoodFormForFoods/EditFoodFormForFoods';
import AddFoodFormToFoods from './AddFoodFormToFoods/AddFoodFormToFoods';

const FoodsDisplayTabs = (props) => (
  <Tabs activeIndex={props.activeIndex}>
    <TabList>
      <Tab>Add</Tab>
      <Tab>Edit</Tab>
    </TabList>

    <TabPanels>
      <TabPanel>
        <AddFoodFormToFoods />
      </TabPanel>
      <TabPanel>
        <EditFoodFormForFoods />
      </TabPanel>
    </TabPanels>
  </Tabs>
);

const mapStateToProps = (state, props) => ({
  activeIndex: state.currentFood.id !== '' ? 1 : 0,
});

export default connect(mapStateToProps)(FoodsDisplayTabs);