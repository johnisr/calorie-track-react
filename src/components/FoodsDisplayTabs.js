import React from 'react';
import { connect } from 'react-redux';
import {Tabs, TabList, Tab, TabPanels, TabPanel} from './Tabs';
import EditFoodFormForFoods from './EditFoodFormForFoods';
import AddFoodFormToFoods from './AddFoodFormToFoods';

const FoodsDisplayTabs = (props) => (
  <Tabs activeIndex={props.activeIndex}>
    <TabList>
      <Tab>Add</Tab>
      <Tab>Edit</Tab>
    </TabList>

    <TabPanels>
      <TabPanel>
        <AddFoodFormToFoods />
      </TabPanel>
      <TabPanel>
        <EditFoodFormForFoods />
      </TabPanel>
    </TabPanels>
  </Tabs>
);

const mapStateToProps = (state, props) => ({
  activeIndex: state.currentFood.id !== '' ? 1 : 0,
});

export default connect(mapStateToProps)(FoodsDisplayTabs);