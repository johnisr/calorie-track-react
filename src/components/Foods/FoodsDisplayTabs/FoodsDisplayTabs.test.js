import React from 'react';
import { shallow } from 'enzyme';
import { FoodsDisplayTabs } from './FoodsDisplayTabs';

it('should render correctly', () => {
  const wrapper = shallow(<FoodsDisplayTabs activeIndex={1} />);
  expect(wrapper).toMatchSnapshot();
});