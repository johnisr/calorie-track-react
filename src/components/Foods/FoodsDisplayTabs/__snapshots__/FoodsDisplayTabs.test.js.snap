// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    activeIndex={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeIndex": 1,
      "children": Array [
        <Unknown>
          <Unknown>
                    Add
          </Unknown>
          <Unknown>
                    Edit
          </Unknown>
</Unknown>,
        <Unknown>
          <Unknown>
                    <Connect(AddFoodFormToFoods) />
          </Unknown>
          <Unknown>
                    <Connect(EditFoodFormForFoods) />
          </Unknown>
</Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown>
              Add
</Unknown>,
            <Unknown>
              Edit
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Add",
            },
            "ref": null,
            "rendered": "Add",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Edit",
            },
            "ref": null,
            "rendered": "Edit",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Unknown>
              <Connect(AddFoodFormToFoods) />
</Unknown>,
            <Unknown>
              <Connect(EditFoodFormForFoods) />
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(AddFoodFormToFoods) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(EditFoodFormForFoods) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeIndex": 1,
        "children": Array [
          <Unknown>
            <Unknown>
                        Add
            </Unknown>
            <Unknown>
                        Edit
            </Unknown>
</Unknown>,
          <Unknown>
            <Unknown>
                        <Connect(AddFoodFormToFoods) />
            </Unknown>
            <Unknown>
                        <Connect(EditFoodFormForFoods) />
            </Unknown>
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown>
                Add
</Unknown>,
              <Unknown>
                Edit
</Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Add",
              },
              "ref": null,
              "rendered": "Add",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Edit",
              },
              "ref": null,
              "rendered": "Edit",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown>
                <Connect(AddFoodFormToFoods) />
</Unknown>,
              <Unknown>
                <Connect(EditFoodFormForFoods) />
</Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Connect(AddFoodFormToFoods) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Connect(EditFoodFormForFoods) />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
